runtime/compiler/optimizer/BoolArrayStoreTransformer.cpp

```diff
@@ -152,8 +152,12 @@ void TR_BoolArrayStoreTransformer::perform()
 
    if (!_bstoreiUnknownArrayTypeNodes->empty())
       {
-      if (_hasByteArrayAutoOrCheckCast && _hasBoolArrayAutoOrCheckCast) // only need to iterate CFG if both byte and boolean array exist
+      if ((_hasByteArrayAutoOrCheckCast && _hasBoolArrayAutoOrCheckCast) || comp()->isDLT())
+         {
+         // need to iterate CFG if both byte and boolean array exist or on DLT compiles
+         // On DLT compiles where type information is not available we must not truncate byte stores to 1-bit
```

Comment:
> we must not truncate byte stores to 1-bit

I think this part of the comment (and the commit message) is misleading. We have to truncate the value at every `bastore` if and only if it's storing to a `boolean[]`. We could just as well incorrectly leave a `bstorei` node unmodified (failing to truncate) when it stores to a `boolean[]` but without any `boolean[]` autos or `checkcast` detected in the trees

I think it would be better to say just that not all of the bytecode has been translated into IL, so there may have been autos and/or `checkcast` operations of both `boolean[]` and `byte[]` type, and the full analysis is needed

Reply:
[Updated the comment](https://github.com/eclipse-openj9/openj9/compare/831e3be7f7fe1974f05967d612b860a1d1437429..1abf31f922dd1c6a4a96c7471667133d09a96b99)